changelogs/fragments/4351-inventory-lxd-handling_metadata_wo_os_and_release.yml

```diff
@@ -0,0 +1,6 @@
+bugfixes:
+  - |
+    lxd inventory plugin - fix handling of metadata to be without os and release.
+    This happens if a empty container is created via API.
+    ```lxc query -X POST /1.0/containers -d '{"name": "container-name", "source": {"type": "none"}, "config": {"security.privileged": "true"}}'```
```

Comment:
What exactly "happens"? Does the plugin crash when such a container is present?
(Also this should be part of the PR description, not of the changelog fragment.)

Reply:
Hello Felix,

I now understand what happened.
The attributes e.g.:
   image.os: ubuntu
   image.release: impish
are not mandatory. The user can delete this information without the container being inoperable. In this case the plugin will crash. I also look at the other attributes. 

Thanks Frank

Reply:
Hi @felixfontein , @conloos .
sorry for my late answer. Like @conloos stated, the metadata that is not mandatory is basically the reason.
The query I used creates a empty container, without any additional information.
Thus os and release are not filled.

Reply:
Sounds good. I'll merge this so this can go into today's extra bugfix release. Further changes can be made in another PR.